File: assp/elem/05_sum_multiples.py
""" Elementary 5:
    Write a program that asks the user for a number
    n and prints the sum of the numbers 1+2+..+n
    but only considers multiples of three or five
    when summing (if three and five are both multiples,
    the number is added to the total once, not twice)
"""

def sum_multiples(N, mults=[3,5]):
    sum_ = 0
    for i in range(1,N+1):
       for m in mults:
          if i % m == 0:
              sum_ += i
              break
    return sum_

N = int(input("Enter a positive integer: "))

assert N > 0, "The entered value must be positive"

print(f"The sum of natural numbers from 1 to {N}",
      f"is {sum_multiples(N)} if considering only multiples of three or five.")

File: assp/elem/05_sum_multiples.c
// Elementary 5: Summation of numbers from 1 to N
//               considering multiples of three and five

#include <stdio.h>
#include <math.h>

int main(void)
{
    unsigned long long int N, sum=0;

    printf("Enter a positive integer: ");
    scanf("%llu", &N);

    for (int i = 1; i <= N; i++)
    {
      if ((i % 3 == 0) || (i % 5 == 0))
      {
          sum += i;
      }
    }

    printf("The sum of natural numbers from 1 to %llu", N);
    printf(" is %llu if considering only multiples of three or five.\n", sum);

    return 0;
}

